src/pages/delivery-orders/Forms/DeliveryOrdersForm.js

```diff
@@ -0,0 +1,762 @@
+          { from: 'seqNo', to: 'mwSeqNo' },
+          { from: 'qty', to: 'qty' },
+          { from: 'functionId', to: 'mwFunctionId' },
+          { from: 'reference', to: 'mwRef' },
+          { from: 'itemId', to: 'itemId' },
+          { from: 'recordId', to: 'mwId' },
+          { from: 'dropDownQty', to: 'deliveredNowQty' },
+          { from: 'dropDownQty', to: 'pendingQty' },
+          { from: 'sku', to: 'sku' },
+          { from: 'doId', to: 'doId' }
+        ],
+        columnsInDropDown: [
+          { key: 'reference', value: 'Reference' },
+          { key: 'itemName', value: 'Name' },
+          { key: 'sku', value: 'SKU' },
+          { key: 'qty', value: 'Qty' }
+        ],
+        displayFieldWidth: 3,
+
+      },
+      propsReducer({ row, props }) {
+        return { ...props, readOnly: !!row.mwId }
+      }
+    },
+    {
+      component: 'textfield',
+      label: labels.sku,
+      name: 'sku',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'textfield',
+      label: labels.itemName,
+      name: 'itemName',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'numberfield',
+      label: labels.qty,
+      name: 'qty',
+      props: {
+        readOnly: isPosted || isCancelled,
+        allowNegative: false
+      },
+      async onChange({ row: { update, oldRow, newRow } }) {
+        if (newRow.qty) {
+          if (newRow.qty > oldRow.qty) {
+            update({
+              qty: oldRow.qty
+            })
+          }
+        }
+      }
+    },
+    {
+      component: 'resourcecombobox',
+      label: labels.site,
+      name: 'siteName',
+      props: {
+        endpointId: InventoryRepository.Site.qry,
+        displayField: 'name',
+        valueField: 'recordId',
+        readOnly: isPosted || isCancelled,
+        mapping: [
+          { from: 'recordId', to: 'siteId' },
+          { from: 'name', to: 'siteName' }
+        ]
+      }
+    }
+  ]
+
+  async function previewBtnClicked() {
+    const data = { printStatus: 2, recordId: formik.values.recordId }
+
+    await postRequest({
+      extension: DeliveryRepository.DeliveriesOrders.flag,
+      record: JSON.stringify(data)
+    })
+
+    invalidate()
+  }
+
+  useEffect(() => {
+    if (documentType?.dtId) formik.setFieldValue('dtId', documentType.dtId)
+  }, [documentType?.dtId])
+
+  function setAddressValues(obj) {
+    Object.entries(obj).forEach(([key, value]) => {
+      formik.setFieldValue(key, value)
+    })
+  }
+
+  function openAddressFilterForm(deliveryOrder) {
+    stack({
+      Component: AddressFilterForm,
+      props: {
+        maxAccess,
+        labels,
+        deliveryOrder,
+        form: formik.values,
+        handleAddressValues: setAddressValues,
+        checkedAddressId: formik.values.addressId
+      },
+      width: 950,
+      height: 600,
+      title: labels.AddressFilter
+    })
+  }
+
+  useEffect(() => {
+    if (!formik.values.clientId) {
```

Comment:
UseEffect is unnecessary here, as this logic can be handled directly within the client's onChange function. If newValue?.recordId is undefined, apply this block of code